plone/scale/storage.py

```diff
@@ -201,11 +201,13 @@ def get_info_by_hash(self, hash):
     def hash_key(self, **parameters):
         if "modified" in parameters:
             del parameters["modified"]
-        key = self.hash(modified=self.modified_time, **parameters)
         fieldname = parameters.get("fieldname", "image")
         dimension = parameters.get("width", parameters.get("scale"))
         if dimension is None:
             dimension = 0
+        if "scale" in parameters:
+            del parameters["scale"]
```

Comment:
I have a doubt here.  How sure are we that we always have a width and height in the parameters?  With all the back and forth between plone.scale and plone.namedfile I don't know if this is the case.
How about the following?  Would that work?

```suggestion
        if "scale" in parameters and parameters.get("width") and parameters.get("height"):
            del parameters["scale"]
```

Reply:
@mauritsvanrees I did a search in the code and realized that all calls to `plone.schema` are made through `plone.namedfile`. And `plone.namedfile` always finds width/height either because it was passed to it or because it was calculated from the scale:

https://github.com/plone/plone.namedfile/blob/6687472bac29807942219a5fde4721f75fdb1132/plone/namedfile/scaling.py#L591-L601

So I don't know if your suggestion would be necessary. Let me know.

Reply:
I agree with  @mauritsvanrees  suggested change. plone.namedfile is not the only consumer of the code. There are addons and custom code as well suing it.

Reply:
@mauritsvanrees @jensens done.